### PR TITLE
support for TLSv1.2

### PR DESCRIPTION
**Pull request by kevinneufeld**

### CODE REVIEW

File: src/Raven.php [hunk 1]
```diff
@@ -246,11 +246,16 @@ public function getRavenRequestString() {
...
-            'content' => $data
-        ));
+        $params = [
+            'http' => [
+                'method' => 'POST',
+                'header' => 'Content-type: application/x-www-form-urlencoded',
+                'content' => $data
+            ],
+            'ssl' => [
+                'crypto_method' => STREAM_CRYPTO_METHOD_TLSv1_2_CLIENT
```
Comment (bslobodin) [inline src/Raven.php]: Is this strictly needed given we enforce 1.2 on the edge? https://github.com/deepcovelabs/raven-sdk-php/blob/0944073ee0f635a7bf0b44906b9cd6f51ea64306/src/Raven.php#L77

Comment (jbeeko) [inline src/Raven.php]: PCI backed off and we don't enforce it except for the .net domain.

Sent from my mobile

On Feb 16, 2016, at 6:30 PM, Boris Slobodin <notifications@github.com<mailto:notifications@github.com>> wrote:


In src/Raven.php<https://github.com/deepcovelabs/raven-sdk-php/pull/10#discussion_r53114133>:

> @@ -246,11 +246,16 @@ public function getRavenRequestString() {
>      */
>      private function postRequest($url, $data, $optional_headers = null) {
>
> -        $params = array('http' => array(
> -            'method' => 'POST',
> -            'header' => 'Content-type: application/x-www-form-urlencoded',
> -            'content' => $data
> -        ));
> +        $params = [
> +            'http' => [
> +                'method' => 'POST',
> +                'header' => 'Content-type: application/x-www-form-urlencoded',
> +                'content' => $data
> +            ],
> +            'ssl' => [
> +                'crypto_method' => STREAM_CRYPTO_METHOD_TLSv1_2_CLIENT


Is this strictly needed given we enforce 1.2 on the edge? https://github.com/deepcovelabs/raven-sdk-php/blob/0944073ee0f635a7bf0b44906b9cd6f51ea64306/src/Raven.php#L77

-

Comment (bslobodin) [inline src/Raven.php]: @jbeeko but the SDK uses `.net` domain, so perhaps we could choose to let it handle the negotiation? Just curious.

Comment (jbeeko) [inline src/Raven.php]: I’m not sure it came across. My feeling is that we should not try force the library to pick any particular version of crypto. The server will only allow negotiation of a valid choice. So there is no need to have the client force anything, which means one less potential configuration headache.

**Pull request merged**